tests/test_with.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-FileCopyrightText: 2023 Henrik Sandklef
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import json
+import pytest
+
+from flict.flictlib.arbiter import Arbiter
+
+arbiter = Arbiter()
+
+def _test_expression(inbound, outbound, result):
+    compats = arbiter.inbounds_outbound_check(outbound, [inbound])
+    assert result == compats['compatibility']
+
+def test_with_1():
+    _test_expression('X11 AND (GPL-2.0-only WITH Classpath-exception-2.0 OR MIT)',
+                     'GPL-2.0-only',
+                     'Yes')
+
+def test_with_2():
+    _test_expression('X11 AND (GPL-2.0-only WITH Classpath-exception-2.0 OR MIT)',
+                     'GPL-3.0-or-later',
+                     'Yes')
+
+def test_with_3():
+    _test_expression('X11 AND (GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-3.0-only)',
+                     'GPL-2.0-only',
+                     'Yes')
+
+def test_with_4():
+    _test_expression('X11 AND (GPL-2.0-only WITH Classpath-exception-2.0 AND GPL-3.0-only)',
+                     'GPL-2.0-only',
+                     'No')
+    
+def test_with_5():
+    _test_expression('X11 AND (GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-3.0-only) AND (X11 OR ISC) AND (BSL-1.0 OR BSD-2-Clause)',
+                     'GPL-2.0-only',
+                     'Yes')
+    
+def test_with_6():
+    _test_expression('X11 AND (GPL-2.0-only WITH Classpath-exception-2.0 AND GPL-3.0-only) AND (X11 OR ISC) AND (BSL-1.0 OR BSD-2-Clause)',
+                     'GPL-2.0-only',
```

Comment:
Shouldn't that be gplv3? I see gplv2 AND gplv3 in the term - so my expectation is gplv3 as the outcome

Reply:
Agree. The test makes sure that is identified by comparing the result against 'No'.